source/common/router/rds_impl.cc

```diff
@@ -99,6 +99,11 @@ void RdsRouteConfigSubscription::onConfigUpdate(
     throw EnvoyException(fmt::format("Unexpected RDS configuration (expecting {}): {}",
                                      route_config_name_, route_config.name()));
   }
+  for (auto* provider : route_config_providers_) {
+    // This seems inefficient, though it is necessary to validate config in each context,
+    // especially when it comes with per_filter_config,
+    provider->validateConfig(route_config);
```

Comment:
@stevenzzzz FYI, keep in mind we need to retain this behavior in the world of SRDS.

Reply:
SG. This is another reason RDS should move to the config-provider-framework. Where RDSSubscription holds a single RDSProvider and this sort of validation(ConfigImpl instantiation) happens before update propagation.

Reply:
> Where RDSSubscription holds a single RDSProvider

@stevenzzzz This is not the case because RDS is being used in multiple context and each context needs its own provider, that's what we address here. We may want to limit that flexibility but it will be hard to do so without breaking API compatibility.